examples/main/main.cpp

```diff
@@ -659,7 +664,15 @@ int main(int argc, char ** argv) {
         // display text
         if (input_echo) {
             for (auto id : embd) {
-                printf("%s", llama_token_to_str(ctx, id).c_str());
+                const char * token_str = llama_token_to_str(ctx, id).c_str();
+                printf("%s", token_str);
```

Comment:
This change results in undefined behavior. AddressSanitizer reports a stack-use-after-scope because the std::string has been freed after line 667. This would work better:
```suggestion
                auto token_str = llama_token_to_str(ctx, id);
                printf("%s", token_str.c_str());
```

Reply:
Thank you for pointing this out.